### ISSUE TITLE
ResultPath: null breaks validator

### ISSUE THREAD

**Issue by mtsvickers** [airware/asl-validator#14]
It is valid syntax to set the ResultPath to null (and is useful sometimes when you don't need the result and don't want it taking up space in your state object). But it breaks the parser. It throws  the following error:
```
{ AssertionError [ERR_ASSERTION]: we need a path null
    at JSONPath.parse (/Users/marissa/Desktop/Local_Sites/stagingpilot-serverless/deployBySfn/node_modules/jsonpath/lib/index.js:16:10)
    at jp.query.map (/Users/marissa/Desktop/Local_Sites/stagingpilot-serverless/deployBySfn/node_modules/asl-validator/src/validator.js:33:12)
    at Array.map (<anonymous>)
    at validator (/Users/marissa/Desktop/Local_Sites/stagingpilot-serverless/deployBySfn/node_modules/asl-validator/src/validator.js:31:6)
    at module.exports (/Users/marissa/Desktop/Local_Sites/stagingpilot-serverless/deployBySfn/src/validateASL.js:29:30)
    at AwsInvokeLocal.invokeLocalNodeJs (/usr/local/lib/node_modules/serverless/lib/plugins/aws/invokeLocal/index.js:331:12)
    at AwsInvokeLocal.invokeLocal (/usr/local/lib/node_modules/serverless/lib/plugins/aws/invokeLocal/index.js:125:19)
    at AwsInvokeLocal.tryCatcher (/usr/local/lib/node_modules/serverless/node_modules/bluebird/js/release/util.js:16:23)
    at Promise._settlePromiseFromHandler (/usr/local/lib/node_modules/serverless/node_modules/bluebird/js/release/promise.js:512:31)
    at Promise._settlePromise (/usr/local/lib/node_modules/serverless/node_modules/bluebird/js/release/promise.js:569:18)
    at Promise._settlePromise0 (/usr/local/lib/node_modules/serverless/node_modules/bluebird/js/release/promise.js:614:10)
    at Promise._settlePromises (/usr/local/lib/node_modules/serverless/node_modules/bluebird/js/release/promise.js:693:18)
    at Async._drainQueue (/usr/local/lib/node_modules/serverless/node_modules/bluebird/js/release/async.js:133:16)
    at Async._drainQueues (/usr/local/lib/node_modules/serverless/node_modules/bluebird/js/release/async.js:143:10)
    at Immediate.Async.drainQueues (/usr/local/lib/node_modules/serverless/node_modules/bluebird/js/release/async.js:17:14)
    at runCallback (timers.js:789:20)
    at tryOnImmediate (timers.js:751:5)
    at processImmediate [as _immediateCallback] (timers.js:722:5)
    generatedMessage: false,
    name: 'AssertionError [ERR_ASSERTION]',
    code: 'ERR_ASSERTION',
    actual: false,
    expected: true,
    operator: '==' }
```
But it also then causes normal tasks to return errors claiming Resource is an additional property or that it isn't a string when it is.
```
{ keyword: 'additionalProperties',
    dataPath: '.States[\'Log Status: Beginning Updates\']',
    schemaPath: 'http://asl-validator.cloud/succeed#/additionalProperties',
    params: { additionalProperty: 'Resource' },
    message: 'should NOT have additional properties' }
```

**Comment by ChristopheBougere** [comment]
Hi @mtsvickers 
Thanks for the bug report.
Could you provide a simple JSON to reproduce it?

**Comment by mtsvickers** [comment]
Sure:

```
{
  "StartAt": "Starting Point",
  "States": {
    "Starting Point": {
      "Type": "Task",
      "Resource": "arn:aws:lambda:us-west-2:1234:function:myservice-dev-myfunction",
      "ResultPath": null,
      "Next": "Simple Task"
    },
    "Simple Task": {
	    "Type": "Task",
	    "ResultPath": "$.myResult",
	    "Resource": "arn:aws:lambda:us-west-2:1234:function:myservice-dev-myfunction",
	    "Next": "End State"
    },
    "End State": {
	    "Type": "Pass",
	    "End": true
    }
  }
}
```

**Comment by ChristopheBougere** [comment]
Can you have a look to #16 ? :)

**Comment by mtsvickers** [comment]
That fixed it! Thank you very much.

**Comment by ChristopheBougere** [comment]
Released in 1.0.7 👍

**Comment by mtsvickers** [comment]
Thank you so much! It's such a wonderful surprise to have someone be so
prompt and responsive. I really appreciate it.
